docs/index.md

```diff
@@ -0,0 +1,1025 @@
+
+**Note:** Snowfakery only works for models which are amenable to having an id column on every record. Your tools can use the column, ignore the column, or exchange it for another kind of ID (as CumulusCI does, with Salesforce) but Snowfakery always generates IDs and refers between tables with IDs. Future versions might include a command line option to turn this behaviour on or off.
+
+Let’s make this example more interesting:
+
+persons_of_interest.yml
+
+```
+- object: Person
+  count: 3
+  fields:
+    name:
+      fake: name
+    age:
+      random_number:
+        min: 12
+        max: 95
+```
+
+What did we say here?
+
+* object: Person : This is a template for rows that will go in the Person table representing real-world “Person” objects
+* count: we want 3 records instead of just 1
+* name: fake: name : we want to fake the name instead of hard coding it. The list of things you can “fake” is based on a library called “Faker” which we will discuss later.
+* age: random_number: we want a random number between the min and max
+
+Now you should get an output more like this:
+
+```
+Person(id=1, name=Allison Garcia, age=94)
+Person(id=2, name=Megan Campos, age=67)
+Person(id=3, name=Katherine Nelson, age=92)
+```
+
+We have created people! Or at least fake personas for people! And every time you run it, you will get a different set of “people”.
+
+So that’s pretty cool, but it doesn’t use much of Snowfakery’s power. Let's go deeper.
+
+pet_stories.yml
+
+```
+- object: Person
+  count: 3
+  fields:
+    name:
+      fake: name
+    age:
+      random_number:
+        min: 12
+        max: 95
+    dog:
+      - object: Animal
+        fields:
+          name:
+            fake: first_name
+          species: canine
+          food:
+            - object: PetFood
+              fields:
+                name: Pets Choice
+                cost: $10
+
+    cat:
+      - object: Animal
+        fields:
+          name:
+            fake: first_name
+          species: feline
+          food:
+            - object: PetFood
+              fields:
+                name: Pets Choice
+                cost: $10
+```
+
+If you’re lost: don't worry! Those are a lot of new concepts at once, and you don't have to understand everything in this example right now. We'll come back to these concepts one by one.
+
+But in case you’re in a hurry: In this case we're creating 3 Person objects. Each one has a name, age, dog and cat. Each dog or cat is an Animal, so we'll get 2 animals per Person or a total of 6. Each animal has a favorite food, so we'll get 6 PetFood objects as well.
+
+<img src='https://quip.com/blob/PJUAAAi0SAB/UReZ0eb4vVO7ebxjxPgLVQ?a=F4goQUMAQOoinOTXGrHGMKYhgKzTZ4BouVaK76LD29Ea' id='PJUACATPCzI' alt='Relationship diagram' width='800' height='190'>Later, we’ll discuss how we could have just 2 PetFood objects which are shared. We’ll also discuss how we could randomly select a Pet species or Food. 
```

Comment:
Are we sure these image links to quip will endure? Should we not include the images in the repository?

Reply:
a8f0672